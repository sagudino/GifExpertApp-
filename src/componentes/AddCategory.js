import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState('');
  const inputChange = (e) => {
    setinputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setinputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        value={inputValue}
        onChange={inputChange}
        placeholder='Busca un gif y pulsta intro...'
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
