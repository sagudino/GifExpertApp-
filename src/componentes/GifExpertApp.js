import React, { useState } from 'react';

export const GifExpertApp = () => {
  //const categories = ['Categoría 1', 'Categoría 2', 'Categoría 3'];
  const [categories, setCategories] = useState([
    'Categoría 1',
    'Categoría 2',
    'Categoría 3',
  ]);

  const anadirCategoria = () => {
    setCategories([...categories, 'Categoría 4']);
  };
  return (
    <div className='container'>
      <h1>GifExpertApp</h1>
      <hr />
      <button onClick={anadirCategoria}>Agregar</button>
      <ul>
        {categories.map((cat) => {
          return <li key={cat}>{cat}</li>;
        })}
      </ul>
    </div>
  );
};
