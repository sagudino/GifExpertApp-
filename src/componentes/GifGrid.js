import React, { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);
  // solo se ejecuta cuando el componente se renderiza la primera vez
  useEffect(() => {
    getGifs(category).then((imgs) => {
      setImages(imgs);
    });
  }, [category]);

  //getGifs();
  return (
    <>
      <h3>{category}</h3>
      <div className='card-grid'>
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
