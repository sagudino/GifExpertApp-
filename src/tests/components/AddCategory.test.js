import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../componentes/AddCategory';

describe('Pruebas en AddCategory', () => {
  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);
  test('Debe mostrarse corectamente', async () => {
    expect(wrapper).toMatchSnapshot();
  });
  /** cambios en el input de bÃºsqueda */
  test('Debe de cambiar la caja de texto', async () => {
    const input = wrapper.find('input');
    const value = 'Hola mundo';
    input.simulate('change', { target: { value: value } });
  });
});
