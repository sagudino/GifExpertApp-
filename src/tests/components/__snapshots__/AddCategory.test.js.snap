// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory Debe mostrarse corectamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Busca un gif y pulsta intro..."
    type="text"
    value=""
  />
</form>
`;
